### ISSUE TITLE
float()-ing a numeral string returns a factually wrong value.

### ISSUE THREAD

**Issue by pajcek11** [godotengine/godot#40624]
**Godot version:**
v3.2.2.stable.official

**OS/device including version:**
Xubuntu 18.4

**Issue description:**
I wanted to turn a string containing numbers into a float

**Steps to reproduce:**
Literally write:
print(float("12314.123123")). it can be any number

Note the second number being factually wrong
![Screenshot](https://user-images.githubusercontent.com/67163678/88265186-8ecdee80-ccbc-11ea-982b-2feee7b7ad29.png)

**Minimal reproduction project:**
Not worth the effort as it can be reproduced in 20 characters

**Comment by Calinou** [comment]
I think this is expected, as the numbers you've specified can't accurately be represented in a 32-bit floating-point number.

**Comment by ThakeeNathees** [comment]
defining `REAL_T_IS_DOUBLE` will give more precision, but why godot is going with 32-bit floats?

```c++
// math_defs.h
#ifdef REAL_T_IS_DOUBLE
typedef double real_t;
#else
typedef float real_t;
#endif
```
32-bit float : 122.54 = 122.54000091552734375
64-bit float : 122.54 = 122.540000000000006252776074689

**Comment by Calinou** [comment]
@ThakeeNathees 32-bit floats are faster compared to 64-bit floats. There are plans to fix `REAL_T_IS_DOUBLE` so it can reliably be used as a build-time option in 4.0, but it won't be the default option for performance reasons (especially on mobile/Web platforms).

**Comment by aaronfranke** [comment]
This isn't related to `REAL_T_IS_DOUBLE` because this isn't about vectors or physics. Variant floats and ints are supposed to be always 64-bit. This wasn't the case for strings, so I fixed it here: #40678.
